### PR TITLE
Fix compiler warnings (-Wsuggest-attribute=noreturn)

### PR DESCRIPTION
**Pull request by fhuberts**
See #784

### CODE REVIEW

File: src/oatpp/core/async/Executor.hpp [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #ifndef oatpp_async_Executor_hpp
 #define oatpp_async_Executor_hpp
 
+#include "../base/Compiler.hpp"
```
Comment (lganzzzo) [inline src/oatpp/core/async/Executor.hpp:28]: `..` isn't allowed in the include paths. 
Please change to "oatpp/core/base/Compiler.hpp"

File: src/oatpp/network/monitor/ConnectionInactivityChecker.hpp [hunk 2]
```diff
@@ -25,6 +25,7 @@
 #ifndef oatpp_network_monitor_ConnectionInactivityChecker_hpp
 #define oatpp_network_monitor_ConnectionInactivityChecker_hpp
 
+#include "../../core/base/Compiler.hpp"
```
Comment (lganzzzo) [inline src/oatpp/network/monitor/ConnectionInactivityChecker.hpp:28]: `..` isn't allowed in the include paths. 
Please change to "oatpp/core/base/Compiler.hpp"

File: src/oatpp/network/monitor/ConnectionMaxAgeChecker.hpp [hunk 3]
```diff
@@ -25,8 +25,10 @@
 #ifndef oatpp_network_monitor_ConnectionMaxAgeChecker_hpp
 #define oatpp_network_monitor_ConnectionMaxAgeChecker_hpp
 
+#include "../../core/base/Compiler.hpp"
```
Comment (lganzzzo) [inline src/oatpp/network/monitor/ConnectionMaxAgeChecker.hpp:28]: `..` isn't allowed in the include paths. 
Please change to "oatpp/core/base/Compiler.hpp"

File: src/oatpp/network/monitor/ConnectionInactivityChecker.hpp [hunk 4]
```diff
@@ -49,7 +50,7 @@ class ConnectionInactivityChecker : public MetricsChecker {
 
   std::vector<oatpp::String> getMetricsList() override;
 
-  std::shared_ptr<StatCollector> createStatCollector(const oatpp::String& metricName) override;
+  std::shared_ptr<StatCollector> createStatCollector(const oatpp::String& metricName) override GPP_ATTRIBUTE(noreturn);
```
Comment (lganzzzo) [inline src/oatpp/network/monitor/ConnectionInactivityChecker.hpp:53]: Doesn't look like a `noreturn` function to me.
Please give more details.

File: src/oatpp/network/monitor/ConnectionMaxAgeChecker.hpp [hunk 5]
```diff
@@ -46,7 +48,7 @@ class ConnectionMaxAgeChecker : public MetricsChecker {
 
   std::vector<oatpp::String> getMetricsList() override;
 
-  std::shared_ptr<StatCollector> createStatCollector(const oatpp::String& metricName) override;
+  std::shared_ptr<StatCollector> createStatCollector(const oatpp::String& metricName) override GPP_ATTRIBUTE(noreturn);
```
Comment (lganzzzo) [inline src/oatpp/network/monitor/ConnectionMaxAgeChecker.hpp:51]: Doesn't look like a `noreturn` function to me.
Please give more details.

File: src/oatpp/core/async/Executor.hpp [hunk 6]
```diff
@@ -63,9 +64,9 @@ class Executor {
 
     oatpp::async::Processor& getProcessor();
 
-    void pushTasks(utils::FastQueue<CoroutineHandle>& tasks) override;
+    void pushTasks(utils::FastQueue<CoroutineHandle>& tasks) override GPP_ATTRIBUTE(noreturn);
```
Comment (lganzzzo) [inline src/oatpp/core/async/Executor.hpp:67]: Doesn't look like a `noreturn` function to me.
Please give more details.

File: src/oatpp/core/async/Executor.hpp [hunk 7]
```diff
@@ -63,9 +64,9 @@ class Executor {
...
 
-    void pushOneTask(CoroutineHandle* task) override;
+    void pushOneTask(CoroutineHandle* task) override GPP_ATTRIBUTE(noreturn);
```
Comment (lganzzzo) [inline src/oatpp/core/async/Executor.hpp:69]: Doesn't look like a `noreturn` function to me.
Please give more details.

**Review by lganzzzo — CHANGES_REQUESTED** [review]

File: src/oatpp/network/monitor/ConnectionMaxAgeChecker.hpp (hunk 5 above)
Comment (fhuberts) [reply src/oatpp/network/monitor/ConnectionMaxAgeChecker.hpp:51]: The implementation throws a std::runtime_error

File: src/oatpp/core/async/Executor.hpp (hunk 6 above)
Comment (fhuberts) [reply src/oatpp/core/async/Executor.hpp:67]: The implementation throws a std::runtime_error

File: src/oatpp/core/async/Executor.hpp (hunk 7 above)
Comment (fhuberts) [reply src/oatpp/core/async/Executor.hpp:69]: The implementation throws a std::runtime_error

File: src/oatpp/network/monitor/ConnectionMaxAgeChecker.hpp (hunk 5 above)
Comment (lganzzzo) [reply src/oatpp/network/monitor/ConnectionMaxAgeChecker.hpp:50]: Yes, you are correct. It always throws.

**Review by lganzzzo — APPROVED** [review]

**Comment by lganzzzo** [comment]
merged

**Pull request merged**